### PR TITLE
[KYUUBI #4053] Support GetTableTypes for trino fe

### PR DESCRIPTION
**Pull request by Yikf**
### _Why are the changes needed?_

We intend to introduce Trino FE at https://github.com/apache/kyuubi/issues/3901, This issue aims to support `GetTableTypes` as subtask.

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by Yikf** [comment]
cc @ulysses-you

File: kyuubi-server/src/main/scala/org/apache/kyuubi/sql/parser/trino/KyuubiTrinoFeAstBuilder.scala [hunk 1]
```diff
@@ -51,4 +51,9 @@ class KyuubiTrinoFeAstBuilder extends KyuubiTrinoFeBaseParserBaseVisitor[AnyRef]
   override def visitGetCatalogs(ctx: KyuubiTrinoFeBaseParser.GetCatalogsContext): KyuubiTreeNode = {
     GetCatalogs()
   }
+
+  override def visitGetTableTypes(ctx: KyuubiTrinoFeBaseParser.GetTableTypesContext)
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/sql/parser/trino/KyuubiTrinoFeAstBuilder.scala:55]: can we import `KyuubiTrinoFeBaseParser._` so all methods in this class can be inline within 100 characters

Comment (Yikf) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/sql/parser/trino/KyuubiTrinoFeAstBuilder.scala]: Thanks, updated

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**